converters/csv_converter.py

```diff
@@ -0,0 +1,192 @@
+import csv
+import os
+
+import pandas as pd
+from babel.numbers import parse_decimal
+
+
+class CSVConverter:
+    def __init__(self, args, input_files: dict):
+        self._input_files = input_files
+        self._xlsx_filename = args.xlsx_filename
+
+        self._current_file = 0
+
+        self._buy_events = []
+        self._sell_events = []
+        self._dividend_events = []
+        self._forex_events = []
+
+        self.df_rsu = pd.DataFrame(
+            columns=[
+                "date",
+                "symbol",
+                "gross_quantity",
+                "net_quantity",
+                "fair_market_value",
+                "currency",
+                "comment",
+            ]
+        )
+        self.df_espp = pd.DataFrame(
+            columns=[
+                "date",
+                "symbol",
+                "buy_price",
+                "fair_market_value",
+                "quantity",
+                "currency",
+                "comment",
+            ]
+        )
+        self.df_deposits = pd.DataFrame(
+            columns=[
+                "date",
+                "symbol",
+                "quantity",
+                "buy_price",
+                "currency",
+                "fees",
+                "fee_currency",
+                "comment",
+            ]
+        )
+        self.df_sales = pd.DataFrame(
+            columns=[
+                "date",
+                "symbol",
+                "quantity",
+                "sell_price",
+                "currency",
+                "fees",
+                "fee_currency",
+                "comment",
+            ]
+        )
+        self.df_dividends = pd.DataFrame(
+            columns=[
+                "date",
+                "symbol",
+                "amount",
+                "tax_withholding",
+                "currency",
+                "comment",
+            ]
+        )
+        self.df_forex = pd.DataFrame(
+            columns=["date",
+                     "source_amount",
+                     "source_currency",
+                     "target_amount",
+                     "target_currency",
+                     "fees",
+                     "fee_currency",
+                     "comment"]
+        )
+        self.df_money_transfers = pd.DataFrame(
```

Comment:
for things like that i started to introduce the "DFRow" classes, technically each sheet should have a corresponding equivalent object representation which also should give you the column names (right now not too transparently and only through they keys of an empty-dict representation)

Reply:
Yes, I know; it's just the rewrite would have taken more time. One additional problem is that I use DF functions to quickly scan through and change the data and the row format is not ideal for this processing. I'll take a look into it nevertheless, but I think I can do it after we merge the change as it won't change the output (it's just internal refactoring)